onnxruntime/core/providers/cpu/tensor/scatter.cc

```diff
@@ -0,0 +1,135 @@
+  const TensorShape& input_data_shape = data_input->Shape();
+  const Tin* indices_data = indecies_input->template Data<Tin>();
+  const auto num_indices = indecies_input->Shape().Size();
+  for (int64_t i = 0; i < num_indices; ++i) {
+    Tin idx = indices_data[i];
+    if (idx < 0 || idx >= input_data_shape[axis]) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "indices element out of data bounds, idx=", idx,
+                             " data_dim=", input_data_shape[axis]);
+    }
+  }
+
+  const auto input_elements = input_data_shape.Size();
+  const auto element_bytes = data_input->DataType()->Size();
+  const auto total_input_bytes = input_elements * element_bytes;
+
+  const uint8_t* src_base = reinterpret_cast<const uint8_t*>(data_input->DataRaw());
+  uint8_t* dst_base = reinterpret_cast<uint8_t*>(data_output->MutableDataRaw());
+  const bool is_string_type = data_input->DataType() == DataTypeImpl::GetType<std::string>();
+
+  // We allow runtime to re-use input for output. If input/output Tensor* are the same
+  // we do not copy
+  if (data_input != data_output) {
+    if (is_string_type) {
+      const std::string* str_begin = data_input->template Data<std::string>();
+      const std::string* str_end = str_begin + input_elements;
+      std::string* dst = data_output->template MutableData<std::string>();
+      std::copy(str_begin, str_end, dst);
+    } else {
+      memcpy(dst_base, src_base, total_input_bytes);
+    }
+  }
+
+  // Now poke updates
+  const auto data_batches = input_data_shape.SizeToDimension(axis);
+  const auto data_batch_bytes = input_data_shape.SizeFromDimension(axis) * element_bytes;
+  const auto block = input_data_shape.SizeFromDimension(axis + 1);
+  const auto block_bytes = block * element_bytes;
+
+  // Initialize with zeros.
+  const uint8_t* update_data = reinterpret_cast<const uint8_t*>(updates_input->DataRaw());
+  for (int64_t batch = 0; batch < data_batches; ++batch) {
+    const auto batch_bytes_offset = batch * data_batch_bytes;
+    for (int64_t update_idx = 0; update_idx < num_indices; ++update_idx) {
+      const Tin axis_idx = indices_data[update_idx];
+      const auto block_offset = (block > 1) ? update_idx % block : 0;
+      const auto dst_offset_bytes = batch_bytes_offset + axis_idx * block_bytes +
+                                    block_offset * element_bytes;
+
+      assert(dst_offset_bytes < total_input_bytes);
+      if (is_string_type) {
+        reinterpret_cast<std::string*>(dst_base)[dst_offset_bytes / element_bytes] =
+            reinterpret_cast<const std::string*>(update_data)[update_idx];
+      } else {
+        // Copy the element
+        auto src_offset_bytes = update_idx * element_bytes;
+        memcpy(dst_base + dst_offset_bytes, update_data + src_offset_bytes, element_bytes);
+      }
+    }
+  }
+  return Status::OK();
+}
+
+Status Scatter::Compute(OpKernelContext* context) const {
+  auto data_input = context->Input<Tensor>(0);
```

Comment:
>auto [](start = 2, length = 4)

Prefer being more explicit with auto and use 'const', '*' and '&' where applicable, so 'const auto*' here.

Reply:
The asterixes are getting converted into start/end italics. 
`const auto *` here